eis_qgis_plugin/processing/algorithms/filtering/focal_filter.py

```diff
@@ -0,0 +1,73 @@
+from qgis.core import (
+    QgsProcessingParameterEnum,
+    QgsProcessingParameterNumber,
+    QgsProcessingParameterRasterDestination,
+    QgsProcessingParameterRasterLayer,
+)
+
+from eis_qgis_plugin.processing.eis_processing_algorithm import EISProcessingAlgorithm
+
+
+class EISFocalFilter(EISProcessingAlgorithm):
+    def __init__(self) -> None:
+        super().__init__()
+
+        self._name = "focal_filter"
+        self._display_name = "Focal filter"
+        self._group = "Filtering"
+        self._group_id = "filtering"
+        self._short_help_string = "Apply a basic focal filter to the input raster"
+
+    def initAlgorithm(self, config=None):
+        self.alg_parameters = [
+            "input_raster",
+            "method",
+            "size",
+            "shape",
+            "output_raster",
+        ]
+
+        input_raster_param = QgsProcessingParameterRasterLayer(
+                name=self.alg_parameters[0], description="Input raster"
+            )
+        input_raster_param.setHelp("The input raster data set.")
+        self.addParameter(input_raster_param)
+       
+        method_param = QgsProcessingParameterEnum(
+            name=self.alg_parameters[1],
+            description="Method",
+            options=["mean", "median"],
+            defaultValue="mean",
+        )
+        method_param.setHelp(
+            "The method to use for filtering. Can be either 'mean' or 'median'. Default to 'mean'."
```

Comment:
I think we could omit the "Defaults to X" parts from the help strings since the defaults show immediately to the user in the GUI. In the toolkit, they are a bit redundant too, but perhaps sometimes helpful if someone is only reading a docstring.